R/bundle.R

```diff
@@ -503,13 +520,50 @@ inferDependencies <- function(appMode, hasParameters, python, hasPythonRmd) {
   unique(deps)
 }
 
-inferPythonEnv <- function(workdir, python) {
+isWindows <- function() {
+  Sys.info()[["sysname"]] == "Windows"
+}
+
+getCondaEnvPrefix <- function(python) {
+  prefix <- dirname(dirname(python))
+  if (!file.exists(file.path(prefix, "conda-meta"))) {
+    stop(paste("Python from", python, "does not look like a conda environment: cannot find `conda-meta`"))
+  }
+  prefix
+}
+
+getCondaExeForPrefix <- function(prefix) {
+  miniconda <- dirname(dirname(prefix))
+  conda <- file.path(miniconda, 'bin', 'conda')
+  if (isWindows()) {
+    conda <- paste(conda, ".exe", sep = "")
+  }
+  if (!file.exists(conda)) {
+    stop(paste("Conda env prefix", prefix, "does not have the `conda` command line interface."))
+  }
+  conda
+}
+
+inferPythonEnv <- function(workdir, python, compatibilityMode, forceGenerate) {
   # run the python introspection script
   env_py <- system.file("resources/environment.py", package = "rsconnect")
-  args <- c(shQuote(env_py), shQuote(workdir))
+  args <- c(shQuote(env_py))
+  if (compatibilityMode || forceGenerate) {
+    flags <- paste('-', ifelse(compatibilityMode, 'c', ''), ifelse(forceGenerate, 'f', ''), sep = '')
+    args <- c(args, flags)
+  }
+  args <- c(args, shQuote(workdir))
 
   tryCatch({
-    output <- system2(command = python, args = args, stdout = TRUE, stderr = NULL, wait = TRUE)
+    if(require('reticulate') && reticulate::py_config()$anaconda && !compatibilityMode) {
```

Comment:
```suggestion
    if (requireNamespace('reticulate', quietly = TRUE) && reticulate::py_config()$anaconda && !compatibilityMode) {
```

Reply:
You might also want to check `reticulate::py_available(initialize = FALSE)` as otherwise calling `reticulate::py_config()` will force `reticulate` to initialize Python.

Reply:
I'm using `'reticulate' %in% installed.packages()` to avoid including reticulate in the namespace where it doesn't belong.